CRM/ManageLetterheads/Page/LetterheadsListPage.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * LetterheadsListPage class.
+ *
+ * A page that displays the full list of letterheads.
+ */
+class CRM_ManageLetterheads_Page_LetterheadsListPage extends CRM_Core_Page {
+
+  /**
+   * @var array
+   *   List of "Available For" labels, indexed by value.
+   */
+  private $availableForLabels;
+
+  /**
+   * Assigns the page's template values.
+   */
+  public function run() {
+    $this->loadAvailableForLabels();
+    $this->assign('letterheads', $this->getLetterheads());
+
+    parent::run();
+  }
+
+  /**
+   * Loads the "Available For" labels, indexed by values.
+   */
+  private function loadAvailableForLabels() {
+    $this->availableForLabels = [];
+    $availableFor = civicrm_api3('OptionValue', 'get', [
+      'sequential' => 0,
+      'option_group_id' => 'manageletterheads_available_for',
+      'options' => [
+        'limit' => 0,
+      ],
+    ]);
+
+    foreach ($availableFor['values'] as $availableFor) {
+      $this->availableForLabels[$availableFor['value']] = $availableFor['label'];
+    };
+  }
+
+  /**
+   * Returns the full list of letterheads.
+   *
+   * The letterheads fields are formatted so they include values as rendered on
+   * the template.
+   *
+   * @return array
+   */
+  private function getLetterheads() {
```

Comment:
We should set up this class to use pagination incase the letterheads list get really long. 
Did something similar here: 

https://github.com/compucorp/rse/blob/master/sites/all/civicrm_extensions/uk.co.compucorp.rse.pomanagement/CRM/PoManagement/Page/PurchaseOrderApprovalList.php#L90-L106

https://github.com/compucorp/rse/blob/master/sites/all/civicrm_extensions/uk.co.compucorp.rse.pomanagement/CRM/PoManagement/Page/PurchaseOrderApprovalList.php#L153-L167

https://github.com/compucorp/rse/blob/master/sites/all/civicrm_extensions/uk.co.compucorp.rse.pomanagement/templates/CRM/PoManagement/Page/PurchaseOrderApprovalList.tpl#L39